paddle/fluid/platform/gpu_info.cc

```diff
@@ -207,6 +207,26 @@ int GetCUDADriverVersion(int id) {
 #endif
   return driver_version;
 }
+std::string GetCUDADeviceName(int id) {
+  PADDLE_ENFORCE_LT(id, GetCUDADeviceCount(),
+                    platform::errors::InvalidArgument(
+                        "Device id must be less than GPU count, "
+                        "but received id is: %d. GPU count is: %d.",
+                        id, GetCUDADeviceCount()));
+
+  PADDLE_ENFORCE_GT(id, -1, platform::errors::InvalidArgument(
+                                "Device id must be greater than -1, "
+                                "but received id is: %d. ",
+                                id));
```

Comment:
I wonder whether `hipGetDeviceProperties` and `cudaGetDeviceProperties` can accept the case `id = -1`. In my experience, they cannot. So I prefer `GetCUDADeviceName` should not accept the case `id = -1`. Users should deal with the case `id = -1` outside this function.

Reply:
Sorry for my wrong understanding. `PADDLE_ENFORCE_GT(id, -1)` is right here. But I prefer `PADDLE_ENFORCE_GE(id, 0)` instead. The latter would be more clear.